### DISCUSSION TITLE
Embedded terminal

### DISCUSSION THREAD

**Discussion by salvatorevassallo** [texstudio-org/texstudio#2821] [Q&A]
Hi
I've seen that it is possible to have an embedded terminal in the lower side of the screen, but I'm not able to obtain it.
I'm running Ubuntu 22.04 and I've tried the usual packages, the development packages and compiling the sources.
When I've compiled the sources I've used the command BUILD.sh and I've answered "yes" to the question if I like the embedded terminal. My qt version is 5.15.3
Can I have some suggestion?
Thank you

**Comment by sunderme** [comment]
you need to install "libqtermwidget5-0-dev"
Build from source via cmake, see file "INSTALL"

**Reply by salvatorevassallo** [reply]
I did it. The compilation was ok and the only warning I had was 

CMake Warning (dev) at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args`
  (QtermWidget) does not match the name of the calling package (QTermWidget).
  This can lead to problems in calling code that expects `find_package`
  result variables (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/FindQTermWidget.cmake:46 (find_package_handle_standard_args)
  CMakeLists.txt:159 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

if I change the line in CMakeLists.txt from QTermWidget to QtermWidget no more warning 
and then
-- Found QtermWidget: /usr/lib/x86_64-linux-gnu/libqtermwidget5.so  
-- Using QTermWidget5

But unfortunately no internal terminal was built

**Comment by sunderme** [comment]
The description on what you did and what did not work is a bit ambiguous.
So, logs/screenshots make things clearer.

What shows you that the internal terminal was not built ?

**Reply by salvatorevassallo** [reply]
I'm sorry. I forgot to uninstall my  old version of texstudio and the menu item opened this version and not the version I compiled. Now all is ok
Thank you for your attention and please forgive me

**Reply by sunderme** [reply]
no problem
